index.js

```diff
@@ -249,9 +249,12 @@ class ThreadStream extends EventEmitter {
     return !this[kImpl].needDrain
   }
 
-  end () {
+  end (cb) {
     if (this[kImpl].destroyed) {
-      throw new Error('the worker has exited')
+      if (typeof cb === 'function') {
+        process.nextTick(cb, new Error('the worker has exited'))
+      }
```

Comment:
Please remove the callback and just return. There is no actual need to throw in this case.